testing/README.md

```diff
@@ -278,6 +278,306 @@ Clears all registered benchmarks, so calling `runBenchmarks()` after it wont run
 them. Filtering can be applied by setting `BenchmarkRunOptions.only` and/or
 `BenchmarkRunOptions.skip` to regular expressions matching benchmark names.
 
+## Behavior-driven development
+
+With the `bdd.ts` module you can write your tests in a familiar format for
+grouping tests and adding setup/teardown hooks used by other JavaScript testing
+frameworks like Jasmine, Jest, and Mocha.
+
+The `describe` function creates a block that groups together several related
+tests. The `it` function registers an individual test case. The `describe` and
+`it` functions have similar call signatures to `Deno.test`, making it easy to
+migrate from using `Deno.test`.
+
+### Hooks
+
+There are 4 types of hooks available for test suites. A test suite can have
+multiples of each type of hook, they will be called in the order that they are
+registered. The `afterEach` and `afterAll` hooks will be called whether or not
+the test case passes. The all hooks will be called once for the whole group
+while the each hooks will be called for each individual test case.
+
+- `beforeAll`: Runs before all of the tests in the test suite.
+- `afterAll`: Runs after all of the tests in the test suite finish.
+- `beforeEach`: Runs before each of the individual test cases in the test suite.
+- `afterEach`: Runs after each of the individual test cases in the test suite.
+
+If a hook is registered at the top level, a global test suite will be registered
+and all tests will belong to it. Hooks registered at the top level must be
+registered before any individual test cases or test suites.
+
+### Focusing tests
+
+If you would like to only run specific individual test cases, you can do so by
+calling `it.only` instead of `it`. If you would like to only run specific test
+suites, you can do so by calling `describe.only` instead of `describe`.
+
+There is one limitation to this when the individual test cases or test suites
+belong to another test suite, they will be the only ones to run within the top
+level test suite.
```

Comment:
Could you provide an example here? It's not obviously clear what this sentence means.

Reply:
If you look at one of the user test examples I have, this issue can be reproduced by adding another test case or describe block that doesn't belong to the User describe block. If you add `.only` to the test for `getAge` that will be the only test case to run within the top level User describe block. The other test cases that do not belong to the User describe block will also still run. currently the only workaround to ensure the user test step is the only one in the file to run is also adding `.only` to the User describe block so that when Deno.test is called, it will be called with the only flag.

I'll still add an example to readme for this limitation. Just thought I'd try explaining it more here first.

The reason this problem exists is because Deno.test is called the moment describe is called, and at that time, it doesn't know if any of it's steps have the only flag. If I had a hook into when tests are about to run, I could delay registration until the last moment, allowing me to determine if Deno.test should be called with only flag based on if one of it's child steps having the only flag.

Reply:
Actually I thought about this more and think I could resolve limitation for test cases registered within the describe callback. It's just the flat style where it wouldn't be known at time of Deno.test call if a step registered after the describe call has the only flag.